pika/adapters/base_connection.py

```diff
@@ -239,7 +241,10 @@ def _get_error_code(self, error_value):
         return None
 
     def _flush_outbound(self):
-        """Call the state manager who will figure out that we need to write."""
+        """write early, if the socket will take the data why not get it out
+        there asap.
+        """
+        self._handle_write()
```

Comment:
@wjps, I agree that 100% non-blocking would be best. From my experience in developing 100% non-blocking I/O integration with OpenSSL a few years ago (in C), I recall that there were a number of gotchas. At the time, I captured the gotchas that I discovered in code comments in https://github.com/openwebos/libpalmsocket/blob/master/src/psl_channel_fsm_crypto.c.